src/power_grid_model/power_grid_model_c/__init__.py

```diff
@@ -0,0 +1,57 @@
+# SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+import platform
+from importlib.resources import files
+from pathlib import Path
+
+
+def get_pgm_dll_path() -> Path:
```

Comment:
i am not sure whether it makes sense to add this to the `__init__.py`. What's wrong with keeping it in a separate file?

in particular, if a developer does `from power_grid_model import power_grid_model_c`, they will import `platform`, `files`, `Path` and `get_pgm_dll_path`. That feels weird to me.

In fact, if we're doing an `__init__.py`, then I would expect to only have the stuff from `dataset`, `serialization` and maybe the `power_grid_core and PGM `CDLL`, but not the `get_pgm_dll_path`